orca-peering/src/main/kotlin/com/netflix/spinnaker/config/PeeringAgentConfiguration.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.config
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder
+import com.netflix.spectator.api.Registry
+import com.netflix.spinnaker.kork.dynamicconfig.DynamicConfigService
+import com.netflix.spinnaker.orca.notifications.NotificationClusterLock
+import com.netflix.spinnaker.orca.peering.PeeringAgent
+import com.netflix.spinnaker.orca.peering.MySqlRawAccess
+import com.netflix.spinnaker.orca.peering.SqlRawAccess
+import org.jooq.DSLContext
+import org.jooq.SQLDialect
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import java.util.concurrent.Executors
+
+@Configuration
+/**
+ * TODO(mvulfson): this needs to support multiple (arbitrary number of) beans / peers defined in config
+ * TODO(mvulfson): prob also worth creating ConfigurationProperties class for the props
+ */
+class PeeringAgentConfiguration {
+  @Bean
+  @ConditionalOnExpression("\${pollers.peering.enabled:false}")
+  fun peeringAgent(
+    jooq: DSLContext,
+    clusterLock: NotificationClusterLock,
+    dynamicConfigService: DynamicConfigService,
+    registry: Registry,
+    @Value("\${pollers.peering.pool-name}") peeredPoolName: String,
+    @Value("\${pollers.peering.id}") peeredId: String,
+    @Value("\${pollers.peering.interval-ms:5000}") pollIntervalMs: Long,
+    @Value("\${pollers.peering.thread-count:200}") threadCount: Int,
+    @Value("\${pollers.peering.chunk-size:100}") chunkSize: Int,
+    @Value("\${pollers.peering.clock-drift-ms:5000}") clockDriftMs: Long
```

Comment:
are we using instance timestamps rather than database timestamps for execution timestamps? I'm not totally sure why we need this

Reply:
(I guess maybe we are instance time-stamping from pre SQL repository days)

Reply:
"MySQL converts TIMESTAMP values from the current time zone to UTC for storage, and back from UTC to the current time zone for retrieval." 

I think you're right as to why millisecond epoch timestamps are still used, but I think the above is a good enough reason to continue down that path. Mostly personal preference (I don't like thinking about or converting TZ's) but bugs can result from the difference in behavior on a locally zoned laptop vs. servers in UTC.

Reply:
We (annoyingly) use instance timestamps for updates (https://github.com/spinnaker/orca/blob/master/orca-sql/src/main/kotlin/com/netflix/spinnaker/orca/sql/pipeline/persistence/SqlExecutionRepository.kt#L652).
I suppose this is something we can change to have MySQL put in the timestamp on insert/update. But until we have that I added this drift thing (which I don't love)
So if there is consensus to use `UNIX_TIMESTAMP()` in our SQL statements, i am all for changing it (in a future PR)